pkg/serverless/trace/cold_start_span_creator.go

```diff
@@ -114,8 +115,8 @@ func (c *ColdStartSpanCreator) createIfReady() {
 }
 
 func (c *ColdStartSpanCreator) create() {
-	// Prevent infinite loop from SpanModifier call
-	if c.lambdaSpan.Name == spanName {
+	// Prevent infinite loop from SpanModifier call and duplicates from timeout restarts
+	if c.lambdaSpan.Name == spanName || c.WasColdStart {
```

Comment:
I'm not seeing where this `WasColdStart` ever gets set to true. Also, this variable name is deceiving. Why would the cold start span creator not want to create a cold start span if it was a cold start? I'd like to see variable names that are more descriptive.

Reply:
Yeah, this does get confusing. The bool is set from [ExecutionContext.SetFromInvocation](https://github.com/DataDog/datadog-agent/blob/e3cd5f8b7f618521fd40095309bd270495b12ada/pkg/serverless/executioncontext/executioncontext.go#L121-L123) and I followed the same naming convention. I guess what this _really_ means is that there was already a cold start in a previous invocation in the current Lambda execution environment. Maybe something more accurate would be ColdStartExistsInExecution

Reply:
Ah gotcha - on subsequent invocations after a timeout, the cold_start_span_creator's `sync.Once.Do()` done counter gets reset due to the execution environment being shut down. So the invocation after a timeout or OOM or other event that triggers a shutdown will create a cold start span

Reply:
Okay, so in that case, the execution environment is shut down. That means the extension is restarted. Does it also mean the user function code is reloaded as well?

Reply:
Yeah, I think that also means the function code is reloaded, but subtly. The invocation after a timeout goes into a ["suppressed init" phase](https://docs.aws.amazon.com/lambda/latest/dg/telemetry-schema-reference.html#InitPhase) and doesn't report any init duration metrics like we would typically find in a cold start.

Reply:
I'm realizing this is a confusing change to make in this PR, so I'm going to create a follow-up PR that will address what I was seeing during my tests with a more descriptive explanation!

Reply:
Awesome, I like this plan! I think that will really help me better understand what you're trying to do here.